### DISCUSSION TITLE
iOS app of UTM in the AppStore?

### DISCUSSION THREAD

**Discussion by 100tomer** [utmapp/UTM#6260] [General]
I saw that know Apple rules change they allow emulation in Europe. Will this mean that UTM can be in the AppStore and maybe without jailbreak?

**Comment by Maxfusion26** [comment]
Will virtualization be available again on the Ipad in the UTM application amid the news that Apple has allowed emulators in the Appstore????
An iPad with Windows arm is the killer of all Snapdragon competitors!

**Comment by Xernium** [comment]
Unfortunately I'll have to burst your bubble
- Apple has removed the virtualization framework and kernel functionality in iOS 16.3.1 and as far as we know they're not planning on adding it back
- As a result you cannot virtualize, only emulate.
- "Fast" emulation requires JIT to be available to the app. There **is** an entitlement that grants apps a form of this, but Apple will seemingly only provide this entitlement to verified / vetted web browsers. And even then, this limited version of JIT is not enough for assisted emulation
- Last option: Slow (unassisted) emulation. Yes, with the new rules this **may** be allowed. But it'll be hard to run anything newer than windows xp with it

**Reply by jthoward64** [reply]
Would still be cool to see UTM SE on the app store for those who cannot use the other options regardless

**Comment by philCryoport** [comment]
FYI UTM SE is now available in the App Store

The Debian 11 ISO on my iPad Pro (2020 gen) is DOG SLOW -- even when booted into text-only mode (i.e. `sudo systemctl set-target multi-user.target`)

**Reply by jthoward64** [reply]
I wonder how an alpine VM would do

**Reply by philCryoport** [reply]
From a freshly downloaded image of UTM's Debian 11 arm64 image, UTM SE took over 6 hours to complete `sudo apt update && sudo apt upgrade -y`

Gonna try jailbreaking my iPad to see if onboard JIT makes a significant difference

**Reply by jthoward64** [reply]
If you jailbreak you can also just run CLI tools directly on the iPad. In theory you can run an XServer over VNC but I don't know that any has really done that for a long time

**Reply by philCryoport** [reply]
Trollstore won't run on iPadOS 17.5.x -- so I can't get the hypervisor

Supposedly the Cydia .ipa has a hypervisor as well, but I can't find if Cydia will run on a jailbroken iPad running 17.5.x

**Reply by Xernium** [reply]
> Trollstore won't run on iPadOS 17.5.x -- so I can't get the hypervisor
> 
> Supposedly the Cydia .ipa has a hypervisor as well, but I can't find if Cydia will run on a jailbroken iPad running 17.5.x

https://github.com/utmapp/UTM/discussions/6260#discussioncomment-9035736

I'll say this again, for Hypervisor you need:
- An Apple device with at least an A12Z CPU (technically ... however do note that UTM's hypervisor publicly only supports M1 or the A16 and newer). Older CPUs simply lack the functionality.
- Running a kernel with hypervisor support (only present in iOS 14-16.3.1)

The first one is a hard requirement. No way around it.
    
The second one is technically not a *hard* requirement but-    
Adding support for this using a Jailbreak is basically impossible.    
**Why?** 
It would require a Jailbreak **more powerful** than any jailbreak before it.   
**A jailbreak based on checkm8 combined with blackbird would not be powerful enough for this even if they worked on CPUs with hypervisor support**    
Reason: Ever since this jailbreak chain, Apple has made enough changes that you would need two more very low level exploits to get around.
**TL;DR:** Unless Apple deems it so, its most most likely not gonna happen in the next decade (if ever)
